### PR TITLE
[WIP] Add 🤗nlp in examples using the updated tokenizer API

### PR DESCRIPTION
**Pull request by thomwolf**
This PR superseded #4864 

This PR examines how to best make use of all the features of 🤗nlp in the examples.

First example studied is GLUE. The main goal is to have explicit data processing (target: no data processing happening inside transformers) as well as add some efficiency features like dynamic/optimized batching. The second goal is to make this a lot more efficient, fast, and reproducible.

### CODE REVIEW

File: examples/text-classification/run_nlp_glue.py [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+        "RobertaTokenizer",
+        "RobertaTokenizerFast",
+        "XLMRobertaTokenizer",
+    ):
+        raise NotImplementedError(
+            """
+        # HACK(label indices are swapped in RoBERTa pretrained model)
+        label_list[1], label_list[2] = label_list[2], label_list[1]
+        """
+        )
```
Comment (julien-c) [inline examples/text-classification/run_nlp_glue.py:201]: Is there a way to re-order label indices in a `nlp` dataset? Or to override a label_classes like is defined in https://github.com/huggingface/nlp/blob/master/datasets/glue/glue.py#L55 ?

If there was we could just remove those hacky lines and pass the label_classes from the config.json when it has a label2ids dict defined.

File: examples/text-classification/run_nlp_glue.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        eval_dataset = glue["validation" + ("_matched" if task_name == "mnli" else "")]
+    if training_args.do_predict:
+        test_dataset = glue["test" + ("_matched" if task_name == "mnli" else "")]
+
+    # Define output mode (regression or classification) and num labels (to define the size of the last layer of the model)
+    output_mode = "regression" if "float" in glue["train"].features["label"].dtype else "classification"
+    if output_mode == "regression":
+        num_labels = 1
+    else:
+        num_labels = len(train_dataset.unique("label"))
```
Comment (julien-c) [inline examples/text-classification/run_nlp_glue.py:189]: this will fail if `do_train` is False, no?

File: examples/text-classification/run_nlp_glue.py (hunk 1 above)
Comment (thomwolf) [reply examples/text-classification/run_nlp_glue.py:201]: good point!
